src/Functions/array/arrayAll.h

```diff
@@ -0,0 +1,86 @@
+#pragma once
+
+#include <Columns/ColumnsNumber.h>
+#include <DataTypes/DataTypesNumber.h>
+#include "FunctionArrayMapped.h"
+
+namespace DB
+{
+namespace ErrorCodes
+{
+    extern const int ILLEGAL_COLUMN;
+}
+
+/** arrayAll(x1,...,xn -> expression, array1,...,arrayn) - is the expression true for all elements of the array.
+  * An overload of the form f(array) is available, which works in the same way as f(x -> x, array).
+  */
+struct ArrayAllImpl
+{
+    static bool needBoolean() { return true; }
+    static bool needExpression() { return false; }
+    static bool needOneArray() { return false; }
+
+    static DataTypePtr getReturnType(const DataTypePtr & /*expression_return*/, const DataTypePtr & /*array_element*/)
+    {
+        return std::make_shared<DataTypeUInt8>();
+    }
+
+    static ColumnPtr execute(const ColumnArray & array, ColumnPtr mapped)
```

Comment:
I know it's extra work but can the implementations be moved to `cpp` files?
It seems a bit too much to move so much code to header files when most of it can still be in `cpp`.